enhancements/authentication/AuthConfig-missing-fields.md

```diff
@@ -0,0 +1,503 @@
+
+**Steps for the Cluster Administrator:**  
+1. Update the `Authentication` CRD to include the custom `discoveryURL`:  
+   ```yaml
+   apiVersion: config.openshift.io/v1
+   kind: Authentication
+   metadata:
+     name: cluster
+   spec:
+     oidcProviders:
+       - issuer:
+           discoveryURL: "https://custom-idp.example.com/.well-known/openid-configuration"
+    ```
+
+#### 2. Configuring `AudienceMatchPolicy` for Token Validation
+**Scenario:**  
+A cluster administrator needs to configure OpenShift to validate the audience (`aud`) claim in JWT tokens using a flexible matching policy.
+
+**Steps:**
+1. Update the authentication CRD with `AudienceMatchPolicy`:
+   ```yaml
+  apiVersion: config.openshift.io/v1
+  kind: Authentication
+  metadata:
+    name: cluster
+  spec:
+    type: OIDC
+    oidcProviders:
+    - name: foo
+      issuer:
+        audienceMatchPolicy: MatchAny
+     ```
+
+#### 3. Custom UID Mapping for Identity Providers
+**Scenario:**  
+A company needs to map the `uid` claim to a custom identifier in order to properly identify users across different OIDC providers.
+
+**Steps:**
+
+1. Update the authentication CRD to include the custom `uid` mapping:
+   ```yaml
+   apiVersion: config.openshift.io/v1
+    kind: Authentication
+    metadata:
+      name: cluster
+    spec:
+      type: OIDC
+      oidcProviders:
+      - name: foo
+        claimMappings:
+          uid:
+            expression: claims.email
+  ```
+-------------------------------------
+```yaml
+    apiVersion: config.openshift.io/v1
+    kind: Authentication
+    metadata:
+      name: cluster
+    spec:
+      type: OIDC
+      oidcProviders:
+      - name: foo
+        claimMappings:
+          uid:
+            claim: email
+```
+
+### 4. Adding Extra Claims for Role-Based Permissions
+**Scenario:**  
+A cluster administrator wants to utilize an external webhook authorizer to process custom claims (such as `role`) from the OIDC token for access control decisions.
+
+**Steps:**
+
+1. Update the authentication CRD to include extra claims:
+   ```yaml
+   apiVersion: config.openshift.io/v1
+   kind: Authentication
+   metadata:
+     name: cluster
+   spec:
+    type: OIDC
+    oidcProviders:
+      - key: 'example.com/tenant'
+        claimMappings:
+          extra:
+            valueExpression: 'claims.tenant'
+  ```
+
+#### 5. Enforcing Token Expiration Limits (Claim Validation)  
+**Scenario:**  
+A security-conscious organization wants to ensure that all OIDC tokens used for authentication have a maximum expiration time of one hour.  
+
+**Steps:**  
+1. Update the authentication CRD to enforce a maximum token expiration time:  
+   ```yaml
+   apiVersion: config.openshift.io/v1
+   kind: Authentication
+   metadata:
+     name: cluster
+   spec:
+    type: OIDC
+    oidcProviders:
+     - expression: 'claims.exp - claims.nbf <= 86400'
```

Comment:
this belongs under `.claimValidationRules`. Additionally, make this value and the message align with the example  (or vice-versa)?